otel/manifest.json

```diff
@@ -0,0 +1,37 @@
+{
+  "display_name": "OTel",
```

Comment:
Is OTel an official name for Open Telemetry?

Reply:
It needs to be `OTel` so that this is identified as the same integration as the one defined in DataDog/dogweb#52117. I don't think this is displayed to users despite the name 😄 From what I can tell after deploying to staging `public_title` is the one that counts